Add persistence exception handling section

psibi opened the pull request:
Let me know if this is not idiomatic. I did a related post regarding this on the mailing list: https://groups.google.com/forum/#!topic/yesodweb/MKEqVZdfam8

snoyberg commented on cookbook/Handling-Persistence-Exception.md:
```diff
@@ -0,0 +1,82 @@
+# Handling exception in persistence
+
+Say, you have a code like this which throws exception:
```
Perhaps "throws _an_ exception"?

snoyberg commented on cookbook/Handling-Persistence-Exception.md:
```diff
@@ -0,0 +1,82 @@
+    deriving Show
+|]
+
+getUser :: MonadIO m => ReaderT SqlBackend m [Entity User]
```
Minor recommendation: `getUsers` instead of `getUser` may be a bit easier to parse for some.

snoyberg commented:
There's actually a problem with this. Once an exception is thrown within a transaction, most drivers will now consider the transaction in an invalid state, and therefore further changes will fail (I think). For example, I think the code you have here will fail for PostgreSQL (though unfortunately I can't easily test that at the moment).

gregwebs commented:
Yes, generally speaking a DB exception must be caught outside the transaction because the DB will not necessarily throw an exception until commit in some cases (in this case imagine another thread doing the same insertion). In this particular example one could use the PersistUnique functionality. In general sometimes (and in this case) you can break up the transaction and catch errors outside it.

psibi commented:
Yes, infact the code doesn't work with the PostgreSQL backend. I have been little occupied in my day job. Will update the PR by this week.

psibi commented:
I have updated the code to handle it in a database like `PostgreSQL`. Have also added a helper function `mapLiftSqlPersistMPool` which is used for executing each transaction. Probably, it can be added to `persistent` if this is the proper way to do it.

gregwebs commented:
Looks better!

snoyberg commented:
LGTM too.

psibi commented:
Thanks for the feedback, merging it.
